### DISCUSSION TITLE
Import/edit DKIM records

### DISCUSSION THREAD

**Discussion by BornTKill** [postalserver/postal#1493] [Feature suggestions]
Dear,

We plan to migrate from postfix/amavis/opendkim configuration to postal.
Is there a way to import/edit DKIM records into postal ?

We have many customers and we cannot impact them with this project.
Best regards.

**Comment by willpower232** [comment]
There is not an import in the UI however most of the data is managed in the database so you may be able to add all your customers and then tweak the dkim records yourself.

**Comment by BornTKill** [comment]
@willpower232 Thank you to confirm this point. I am going to manage it like this.

**Comment by dragoangel** [comment]
@BornTKill you can have as many dkims as how many  stars on night sky while their selector is unique. What an issue to add needed record to DNS?) I think implement such option to manage dkim secrets over UI is can be security issue.

**Comment by BornTKill** [comment]
@dragoangel It’s not a technical problem but an organisation one. The point is that we have customers and we would like not to bother them with our migration.

However we tried to import existing dkim with DB update but because of dkim_identifier we failed. Even with dkim_identifer empty, the record stay -XxXx.domain.com : the hyphen remains unfortunately.

So we will contact our 50 customers to add new records.

**Comment by dragoangel** [comment]
Oh. Clean now.

**Comment by willpower232** [comment]
@BornTKill thanks for updating us, I can see that would be annoying, I feel like a small change here would resolve it https://github.com/postalhq/postal/blob/ed7889a40e1d777f1d80821043562ab18b04766e/app/models/domain.rb#L122

I wouldn't be sure how officially supported the absence of `dkim_identifier` should be but for your setup, it should be fine.

**Comment by BornTKill** [comment]
Hey!

Thank I found this also but don’t wanna modify original source to avoid future upgrade issue 👍 

Regards.

**Comment by willpower232** [comment]
Fair enough, I'm going to leave this open in case this minor change appeals to others.

**Comment by ghost** [comment]
I'm needing something like this as well, my DNS provider puts everything lowercase and Postal keeps saying the records can't be found. I even copy/pasted the value to make sure it wasn't a typo, even tried a week later just in case propogation didn't take place. I'm not sure if it's supposed to be case sensitive or not, but Postal has it with capital letters.

**Reply by willpower232** [reply]
Please open a fresh discussion with some screenshots to demonstrate what is happening rather than replying to an ancient discussion. I don't think an import would help here.
